liquidity-pool-changes/src/agent.spec.js

```diff
@@ -0,0 +1,234 @@
+const BigNumber = require('bignumber.js');
+
+// pool mocking
+const mockToken0Address = '0xFAKETOKEN0ADDRESS'; // .token0()
+const mockToken1Address = '0xFAKETOKEN1ADDRESS'; // .token1()
+const mockPoolAddress = '0xFAKEPOOLADDRESS'; // .address
+const mockDecimals = 3;
+
+const mockPoolContract = {
+  // balanceOf gets mocked within each test case for modularity
+  token0: jest.fn().mockResolvedValue(mockToken0Address),
+  token1: jest.fn().mockResolvedValue(mockToken1Address),
+  decimals: jest.fn().mockResolvedValue(mockDecimals),
+  address: mockPoolAddress,
+};
+
+// combine the mocked provider and contracts into the ethers import mock
+jest.mock('forta-agent', () => ({
+  ...jest.requireActual('forta-agent'),
+  getEthersProvider: jest.fn(),
+  ethers: {
+    ...jest.requireActual('ethers'),
+    Contract: jest.fn().mockReturnValue(mockPoolContract),
+  },
+}));
+
+const { FindingType, FindingSeverity, Finding } = require('forta-agent');
+
+// axios mocking
+const mockCoinGeckoData = {};
+const mockCoinGeckoResponse = {
+  data: mockCoinGeckoData,
+};
+
+jest.mock('axios', () => ({
+  get: jest.fn().mockResolvedValue(mockCoinGeckoResponse),
+}));
+
+const axios = require('axios');
+
+const { provideHandleBlock, provideInitialize } = require('./agent');
+
+// axios mock test
+describe('mock axios GET requests', () => {
+  it('should call axios.get and return the mocked response for CoinGecko', async () => {
+    mockCoinGeckoResponse.data = { '0xtokenaddress': { usd: 1000 } };
+    const response = await axios.get('https://url.url');
+    expect(axios.get).toHaveBeenCalledTimes(1);
+    expect(response.data['0xtokenaddress'].usd).toEqual(1000);
+
+    // reset call count for next test
+    axios.get.mockClear();
+    expect(axios.get).toHaveBeenCalledTimes(0);
+  });
+});
+
+// handler tests
+describe('large liquidity pool change agent', () => {
+  describe('handleBlock', () => {
+    let initializeData;
+    let handleBlock;
+
+    beforeEach(async () => {
+      initializeData = {};
+
+      await provideInitialize(initializeData)();
+
+      handleBlock = provideHandleBlock(initializeData);
+    });
+
+    it('returns empty findings if liquidity change is below given threshold', async () => {
+      const mockToken0Amount = new BigNumber(10);
+      const mockToken1Amount = new BigNumber(5);
+      const mockToken0Amount2 = new BigNumber(10);
+      const mockToken1Amount2 = new BigNumber(5);
+
+      // used to scale response
+      const decimalScaling = (new BigNumber(10)).pow(mockDecimals);
+
+      // mock coin gecko response data
+      mockCoinGeckoResponse.data = {};
+      mockCoinGeckoResponse.data[mockToken0Address.toLowerCase()] = { usd: new BigNumber(1) };
```

Comment:
Mocked calls to the CoinGecko API should return a Number and not a BigNumber, since that is what the real call will return:
```suggestion
      mockCoinGeckoResponse.data[mockToken0Address.toLowerCase()] = { usd: 1 };
```

This should be changed in each test where the CoinGecko call is mocked out